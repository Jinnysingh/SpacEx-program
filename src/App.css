.App {
  text-align: center;
}
.pt-20{
  padding-top: 20px;
}
.pb-20{
  padding-bottom: 20px;
}
.mt-20{
  margin-top: 20px;
}
.launch-card{
  box-shadow: 0 0 4px 0 rgba(0, 0, 0, 0.14);
  padding: 15px;
}
.tiles{
  height: 100%;
}
.title{
  width: 100%;
}
.launch-btn{
  width: 30%;
  margin-left: 20px;
  margin-right: 20px;
  margin-top: 15px;
  background-color: lightgreen !important;
  border-color: lightgreen !important;
}
.container{
  max-width: 1280px!important;
}
.card-header{
  padding: 0px !important;
  border-bottom: none !important;
}
.launch-img{
  width: 100%;
  height: 100%;
}
.mission-name{
  color: darkblue;
    padding-top: 10px;
    text-align: left;
    padding-bottom: 10px;
    font-size: 14px;
    font-weight: bold;
}
.card-body{
  padding: 0px !important
}
.launch-title{
  font-size: 14px;
  font-weight: bold;
}
.launch-title-right{
  font-size: 14px;
  padding-left: 5px;
  padding-right: 5px;
}
.launch-sec{
  padding-top: 20px;
  padding-bottom: 20px;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
}
.active-btn:focus{
background-color: darkgreen !important;
color: white !important;
border-color: darkgreen !important;
}