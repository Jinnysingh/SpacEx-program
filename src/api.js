const fetch = require('node-fetch');

export const allData = fetch("https://api.spacexdata.com/v3/launches?limit=100")
.then((data)=>{
    return data
})
.catch(err =>{
    console.error("error in allData api calling", err)
})

export const successLaunchFilter = (value, type) => fetch(`https://api.spaceXdata.com/v3/launches?limit=100&amp;${type}=${value}`)
.then((data)=>{
    return data
})
.catch(err =>{
    console.error("error in successLaunchFilter api calling", err)
})

export const successLaunchLand = (launch_success, land_success) => fetch(`https://api.spaceXdata.com/v3/launches?limit=100&amp;launch_success=${launch_success}&amp;land_success=${land_success}`)
.then((data)=>{
    return data
})
.catch(err =>{
    console.error("error in successLaunchLand api calling", err)
})

export const allFilters = (launch_success, land_success, launch_year) => fetch(`https://api.spaceXdata.com/v3/launches?limit=100&amp;launch_success=${launch_success}&amp;land_success=${land_success}&amp;launch_year=${launch_year}`)
.then((data)=>{
    return data
})
.catch(err =>{
    console.error("error in allFilters api calling", err)
})