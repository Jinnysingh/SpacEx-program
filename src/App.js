import React, {useEffect, useState} from "react";
import LaunchFilter from "./LaunchFilter/LaunchFilter.jsx";
import LaunchCard from './LaunchCards/LaunchCards.jsx'
import LazyLoad from 'react-lazyload';
import {allData} from './api.js'
import {successLaunchFilter, successLaunchLand, allFilters} from "./api.js"
import "./App.css";

function App() {
  const [allCard, setCards] = useState([])
  const [launchSuccess, setLaunchSuccess] = useState(null)
  const [landSuccess, setLandSuccess] = useState(null)
  const [launchYear, setLaunchYear] = useState(0)

  useEffect(() => {
    allData
      .then(res => res.json())
      .then(
        (result) => {
          setCards(result)
        },
        (error) => {
          console.error(error)        }
      )
  }, [])

  const handleFilters = (event, type) => {
    let value = event.target.innerText;
    let data = value === "True" ? true : false;
    if (type === "Launch Success") {
      setLaunchSuccess(data);
      if(landSuccess === null && launchYear === 0){
      successLaunchFilter(data)
        .then((res) => res.json())
        .then(
          (result) => {
            setCards(result);
          },
          (error) => {
            console.error(error);
          }
        );
    }
    } else if(type === "Land Success"){
      setLandSuccess(data)
      if(launchSuccess !== null){
      successLaunchLand(launchSuccess, data)
      .then((res) => res.json())
        .then(
          (result) => {
            setCards(result);
          },
          (error) => {
            console.error(error);
          }
        );
      }
    }else if(type === "Launch Years"){
      setLaunchYear(parseInt(value))
      if(launchSuccess && landSuccess !== null){
      allFilters(launchSuccess, landSuccess, launchYear )
      .then((res) => res.json())
        .then(
          (result) => {
            setCards(result);
          },
          (error) => {
            console.error(error);
          }
        );
      }
    }
  };

  return (
    <div className="App">
      <div className="container-fluid">
        <h4 className="text-left pt-20 pb-20">SpacEx Launch program</h4>
        <div className="row">
          <div className="col-md-3">
            <LaunchFilter handleFilters={handleFilters}/>
          </div>
          <div className="col-md-9">
          <LazyLoad height={200} offset={100} >
          <LaunchCard cardDetails={allCard}/>
          </LazyLoad>
          </div>
        </div>
      </div>
    </div>
  );
}

export default App;
