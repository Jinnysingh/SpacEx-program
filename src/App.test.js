import {shallow} from "enzyme"
import toJSON from "enzyme-to-json"
import React from "react"
import App from "./App.js"
import {successLaunchFilter, successLaunchLand, allFilters} from "./api.js"

test("testing the App component", ()=>{
    const wrapper = shallow(<App />)
    expect(wrapper.instance().state.loader).toBe(true);
    expect(toJSON(wrapper)).toMatchSnapshot();
})

test("useEffect is called on update", () => {
    const mockUseEffect = jest.fn();
    React.useEffect = mockUseEffect;
    const wrapper = shallow(<App />);
    mockUseEffect.mockClear();
    wrapper.update();
  });

test('the successLaunchFilter Api test', () => {
    function callback(data) {
      expect(data).toBe(1);
    }
    successLaunchFilter(callback);
  });

  test('the successLaunchLand Api test', () => {
    function callback(data) {
      expect(data).toBe(1);
    }
    successLaunchLand(callback);
  });

  test('the allFilters Api test', () => {
    function callback(data) {
      expect(data).toBe(1);
    }
    allFilters(callback);
  });