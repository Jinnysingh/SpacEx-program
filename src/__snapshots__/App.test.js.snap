// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> testing the App component 1`] = `
<div
  className="App"
>
  <div
    className="container-fluid"
  >
    <h4
      className="text-left pt-20 pb-20"
    >
      SpacEx Launch program
    </h4>
    <div
      className="row"
    >
      <div
        className="col-md-3"
      >
        <LaunchFilter
          handleFilters={[Function]}
        />
      </div>
      <div
        className="col-md-9"
      >
        <LazyLoad
          classNamePrefix="lazyload"
          height={200}
          offset={100}
          once={false}
          overflow={false}
          resize={false}
          scroll={true}
          unmountIfInvisible={false}
        >
          <LaunchCard
            cardDetails={Array []}
          />
        </LazyLoad>
      </div>
    </div>
  </div>
</div>
`;

exports[`testing the App component 1`] = `
<div
  className="App"
>
  <div
    className="container-fluid"
  >
    <h4
      className="text-left pt-20 pb-20"
    >
      SpacEx Launch program
    </h4>
    <div
      className="row"
    >
      <div
        className="col-md-3"
      >
        <LaunchFilter
          handleFilters={[Function]}
        />
      </div>
      <div
        className="col-md-9"
      >
        <LazyLoad
          classNamePrefix="lazyload"
          height={200}
          offset={100}
          once={false}
          overflow={false}
          resize={false}
          scroll={true}
          unmountIfInvisible={false}
        >
          <LaunchCard
            cardDetails={Array []}
          />
        </LazyLoad>
      </div>
    </div>
  </div>
</div>
`;
